egret/model_library/unit_commitment/params.py

```diff
@@ -734,11 +734,16 @@ def fuel_supply_gens_init(m):
             thermal_gen_attrs['fuel_supply'] = dict()
         if 'aux_fuel_supply' not in thermal_gen_attrs:
             thermal_gen_attrs['aux_fuel_supply'] = dict()
-        gen_set = set(thermal_gen_attrs['fuel_supply'].keys())
-        gen_set.update(thermal_gen_attrs['aux_fuel_supply'].keys())
-        return gen_set
+        fuel_supply = thermal_gen_attrs['fuel_supply']
+        for g in fuel_supply:
+            yield g
+        for g in thermal_gen_attrs['aux_fuel_supply']:
+            if g not in fuel_supply:
+                yield g
 
     def gen_cost_fuel_validator(m,g):
+        if g is None:
```

Comment:
It's not immediately clear to me why the generator could be None - can you add a comment, as this seems likely to confuse more broadly.

Reply:
Comment added -- I'm not sure if this is because we use a generator to initialize this set, but apparently this is a case that needs to be caught!